generative_data_prep/utils/metadata_generation.py

```diff
@@ -0,0 +1,179 @@
+from transformers import GPT2Config
+
+
+class DatasetMetadata(BaseModel):
+    """Pydantic model to validate parameters for metadata.
+
+    Args:
+        max_seq_length: max sequence length of model
+        token_type_ids: whether or not contains token type ids (always true)
+        vocab_size: vocab size of tokenizer
+        model_type: model type used for tokenizer
+        number_of_training_files: number of hdf5 files available for training
+        number_of_dev_files: number of hdf5 files available for evaluation during training
+        number_of_test_files: number of test files
+        max_batch_size_train: the minimum datapoints across all training file
+        which is what the maximum batch size during training can be
+        max_batch_size_dev: the minimum datapoints across all dev file which
+        is what the maximum batch size during training can be
+
+    """
+
+    max_seq_length: int
+    token_type_ids: bool
+    vocab_size: int
+    tokenizer_model_type: str
+    number_of_training_files: int
+    number_of_dev_files: Optional[int]
+    number_of_test_files: Optional[int]
+    max_batch_size_train: int
+    max_batch_size_dev: Optional[int]
+
+    @field_validator("max_seq_length")
+    @classmethod
+    def validate_max_seq_length(cls, v: int, info: ValidationInfo):
+        """Validates max sequence length."""
+        runtime_max_seq_length = info.context.get("max_seq_length")
+        if type(runtime_max_seq_length) is not int:
+            raise ValueError("Max sequence length context param should be an integer variable")
+        if v != runtime_max_seq_length:
+            raise ValueError(
+                (
+                    f"max_seq_length specified during training ({runtime_max_seq_length}) "
+                    f"must match max_seq_length used during generative data prep ({v})"
+                )
+            )
+        return v
+
+    @field_validator("number_of_training_files")
+    @classmethod
+    def validate_number_of_training_files(cls, v: int, info: ValidationInfo):
+        """Validates number of training files."""
+        number_of_instances = info.context.get("world_size")
+        if type(number_of_instances) is not int:
+            raise ValueError("World size context param should be an integer variable")
+        if v < number_of_instances:
+            raise ValueError(
+                (
+                    f"The number of workers({number_of_instances}) is greater than "
+                    f"the specified number of files ({v})"
+                )
+            )
+        return v
+
+    @field_validator("vocab_size")
+    @classmethod
+    def validation_vocab_size(cls, v: int, info: ValidationInfo):
+        """Validates vocab size."""
+        runtime_vocab_size = info.context.get("vocab_size")
+        if type(runtime_vocab_size) is not int:
+            raise ValueError("Vocab size context param should be an integer variable")
+        if v > runtime_vocab_size:
```

Comment:
greater than or equal to?

Reply:
or it should be ok to have runtime vocab size = dataset vocab size correct?

Reply:
I am dumb! this looks good